### DISCUSSION TITLE
Is there a way to know which port the client connected?

### DISCUSSION THREAD

**Discussion by fuqifacai** [neutrinolabs/xrdp#3326] [General]
If allow Xrdp server open many ports to allow connection .
Is that a way to know or detect which port of one of clients is connected ?
Or similar way to detect the session of each connection?
Thanks for any advice.

**Comment by matt335672** [comment]
I can't see a simple way to do that at the moment I'm afraid.

**Reply by fuqifacai** [reply]
Is there a way to know which client connect to the server side  from server envrionment , 
since client login in via username and password , the server side should it ,right ?
Thanks for any info.

**Reply by matt335672** [reply]
In the `devel` stream, we've merged utmp/wtmp support - see #2745 

This will let you use the `who` command to see which user is mapped to which X server, and the client that they initially connected from. Note that if a client disconnects and reconnects from another location, the information from `who` will not be updated.
